sklearn/preprocessing/imputation.py

```diff
@@ -115,12 +204,16 @@ class Imputer(BaseEstimator, TransformerMixin):
       contain missing values).
     """
     def __init__(self, missing_values="NaN", strategy="mean",
-                 axis=0, verbose=0, copy=True):
+                 axis=0, verbose=0, copy=True, n_neighbors=10,
+                 row_max_missing=0.5, col_max_missing=0.8):
```

Comment:
`col_max_missing` isn't really part of the KNN imputation feature (except when `axis=1`, which is a weird case). Raising an error in general would break current behaviour. Perhaps this deserves a warning, not an error; and I think it belongs in a separate PR.

Avoiding using largely-missing rows in KNN is a sensible feature, and differs from the whole-dataset statistics behaviour. It may even be sufficient to justify a separate class to KNN.